azkaban-exec-server/src/main/java/azkaban/execapp/AzkabanExecutorServer.java

```diff
@@ -265,23 +265,29 @@ private void startReportingExecMetrics() {
 
   private void initActive() throws ExecutorManagerException {
     final Executor executor;
-    try {
-      final String host = requireNonNull(getHost());
-      // TODO "pick any free port" can't be supported like this because it's only available after the
-      // jetty server has been started. this works if executor.port is set
-      final int port = props.getInt(ConfigurationKeys.EXECUTOR_PORT, -1);
-      checkState(port != -1);
-      executor = this.executionLoader.fetchExecutor(host, port);
-    } catch (final ExecutorManagerException e) {
-      logger.error("Error fetching executor entry from DB", e);
-      throw e;
-    }
-    if (executor == null) {
-      logger.info("This executor wasn't found in the DB. Setting active=false.");
-      getFlowRunnerManager().setActiveInternal(false);
+    final String host = requireNonNull(getHost());
```

Comment:
NPE. Should be inside the try catch block.

Reply:
> NPE. Should be inside the try catch block.

"the try catch block" – do you mean the one below? Or a new one?

I'll move it inside the if block below. Indeed host is not needed here if port is not set.